src/rpc/connection.rs

```diff
@@ -554,40 +557,21 @@ impl CircularQueue {
         rsp_meta_data: &mut [u8],
         rsp_data: &mut [u8],
     ) -> Result<u32, Box<dyn std::error::Error>> {
-        let id = self.end_index.fetch_add(1, Ordering::Relaxed);
+        let idx = self.end_index.fetch_add(1, Ordering::Relaxed);
+        let id = idx % REQUEST_QUEUE_LENGTH as u32;
```

Comment:
Pls make sure that integer overflow issues do not affect.

Reply:
change the mod length will be ok.
<img width="723" alt="Screen Shot 2023-01-30 at 12 38 56" src="https://user-images.githubusercontent.com/82888636/215388789-d01002c5-69ef-4398-919e-460bd30e4b1b.png">

Reply:
Great!